### PR TITLE
PHOENIX-6349 Add and use commons-cli to phoenix-thirdparty

### PR DESCRIPTION
**Pull request by stoty**
includes a patch that lets us disable the double quote removing "feature"

### CODE REVIEW

File: phoenix-shaded-commons-cli/src/main/patches/CLI-254-1.4.patch [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
++
+     public CommandLine parse(Options options, String[] arguments) throws ParseException
+     {
+         return parse(options, arguments, null);
+@@ -232,7 +269,7 @@ public class DefaultParser implements CommandLineParser
+         }
+         else if (currentOption != null && currentOption.acceptsArg() && isArgument(token))
+         {
+-            currentOption.addValueForProcessing(Util.stripLeadingAndTrailingQuotes(token));
++            currentOption.addValueForProcessing(conditionallyStripLeadingAndTrailingQuotes(token));
```
Comment (virajjasani) [inline phoenix-shaded-commons-cli/src/main/patches/CLI-254-1.4.patch:56]: This gets applied when we build phoenix-thirdparty?

**Review by virajjasani — APPROVED** [review]
Just a small question around patch generation that i am not much aware of, else +1 (non-binding)

File: phoenix-shaded-commons-cli/src/main/patches/CLI-254-1.4.patch (hunk 1 above)
Comment (virajjasani) [reply phoenix-shaded-commons-cli/src/main/patches/CLI-254-1.4.patch:56]: Also, 254 in CLI-254 is somehow relevant in commons-cli or is it just a number that we provide in patch file?

Comment (stoty) [reply phoenix-shaded-commons-cli/src/main/patches/CLI-254-1.4.patch:56]: Yes. The pom was copied from hbase-thirdparty:hbase-shaded-protobuf with minimal changes.

The patch is a backport of my proposed fix for https://issues.apache.org/jira/browse/CLI-254
https://github.com/apache/commons-cli/pull/58

 to commons-cli 1.4

File: phoenix-shaded-commons-cli/pom.xml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <filesets>
+            <fileset>
+              <directory>${basedir}/src/main/java</directory>
+              <includes>
+                <include>**/**</include>
+              </includes>
```
Comment (joshelser) [inline phoenix-shaded-commons-cli/pom.xml:58]: Isn't this implied?

File: phoenix-shaded-commons-cli/pom.xml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        </configuration>
+      </plugin>
+      <plugin>
+        <!--Download our dependency src, i.e. commons-cli, and
+             unpack it. Overlays src/main/java so ready for
+            compile-time (the jar plugin expects src in
+            src/main/java)-->
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>3.1.1</version>
```
Comment (joshelser) [inline phoenix-shaded-commons-cli/pom.xml]: Would be good to pull versions into pluginManagement in the parent in a follow-on.

**Review by joshelser — APPROVED** [review]
Looks great. A couple comments, but nothing that needs to be addressed immediately.

File: phoenix-shaded-commons-cli/pom.xml (hunk 3 above)
Comment (stoty) [reply phoenix-shaded-commons-cli/pom.xml]: I'll update it in a few minutes.

File: phoenix-shaded-commons-cli/pom.xml (hunk 2 above)
Comment (stoty) [reply phoenix-shaded-commons-cli/pom.xml:58]: You're right.
I left it in when removing the exclusions when copying this from hbase-shaded-porotobuf.

**Comment by stoty** [comment]
Thanks for the reviews @virajjasani @joshelser

**Pull request closed without merging**